File: engine/engine-model/src/main/scala/net/server/channel/packet/interaction/VisitPlayerInteractionPacket.scala
package net.server.channel.packet.interaction

class VisitPlayerInteractionPacket(private var _mode: Byte, private var _objectId: Int, private var _password: String) extends BasePlayerInteractionPacket(_mode = _mode) {
  def objectId: Int = _objectId

  def password: String = _password

  def this(mode: Byte, objectId: Int) {
    this(mode, objectId, "")
  }
}

File: engine/engine-model/src/main/java/net/server/channel/packet/interaction/VisitPlayerInteractionPacket.java
package net.server.channel.packet.interaction;

public class VisitPlayerInteractionPacket extends BasePlayerInteractionPacket {
   private final Integer objectId;

   private final String password;

   public VisitPlayerInteractionPacket(Byte mode, Integer objectId, String password) {
      super(mode);
      this.objectId = objectId;
      this.password = password;
   }

   public VisitPlayerInteractionPacket(Byte mode, Integer objectId) {
      super(mode);
      this.objectId = objectId;
      this.password = "";
   }

   public Integer objectId() {
      return objectId;
   }

   public String password() {
      return password;
   }
}
